### PR TITLE
Getting map loading in place

### PR DESCRIPTION
**Pull request by ZackMattor**
- Load maps from a file and store them in a Map struct
 - Render the stored maps with a debug renderer

### CODE REVIEW

**Comment by ZackMattor** [comment]
image

`#` is a wall
` ` is the floow
`S` is the player start location
`?` is an unknown tile type

File: src/maps.rs [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+impl From<&String> for Map {
+    fn from(str_data: &String) -> Self {
+        let mut data : Vec<Vec<TileType>> = vec![vec![]];
+        let mut player_start = Point {
+            x: 0,
+            y: 0,
+        };
+
+        for (_i, c) in str_data.chars().enumerate() {
```
Comment (dean-martin) [inline src/maps.rs]: If you're not going to use the i var, you can just replace it with just `_`

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,2 @@
+/target
+**/*.rs.bk
```
Comment (dean-martin) [inline .gitignore:2]: lol
.bk files?

**Comment by dean-martin** [comment]
I'd recommend installing https://github.com/rust-lang/rust.vim and enabling auto fmt with 
`let g:rustfmt_autosave = 1`
Or :RustFmt if you want it on demand

**Comment by dean-martin** [comment]
10/10, approved.

File: .gitignore (hunk 2 above)
Comment (ZackMattor) [reply .gitignore:2]: 🤷 https://github.com/rust-lang/cargo/pull/2409

File: src/maps.rs (hunk 1 above)
Comment (ZackMattor) [reply src/maps.rs]: Turns out `std::str::Chars` is already an iterator :D 

 https://doc.rust-lang.org/std/str/struct.Chars.htm

**Pull request merged**